lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -44,17 +45,79 @@ def run
       def visit_class(node)
         class_name = node.constant.constant.value
         if class_name.end_with?("Test")
-          add_code_lens(node, [@path, class_name, BASE_COMMAND + " -Itest " + @path])
+          add_code_lens(node, name: class_name, command: BASE_COMMAND + @path)
+        end
+        visit(node.bodystmt)
+      end
+
+      sig { override.params(node: SyntaxTree::DefNode).void }
+      def visit_def(node)
+        if @modifier == "public"
+          method_name = node.name.value
+          if method_name.start_with?("test_")
+            add_code_lens(
+              node,
+              name: method_name,
+              command: BASE_COMMAND + @path + " --name " + method_name,
+            )
+          end
+        end
+      end
+
+      sig { override.params(node: SyntaxTree::Command).void }
+      def visit_command(node)
+        if node.message.value == "public"
+          modifier = @modifier
+          @modifier = "public"
+          visit(node.arguments)
+          @modifier = modifier
+        end
+      end
+
+      sig { override.params(node: SyntaxTree::CallNode).void }
+      def visit_call(node)
+        ident = node.message if node.message.is_a?(SyntaxTree::Ident)
+
+        if ident
+          if T.cast(ident, SyntaxTree::Ident).value == "public"
+            modifier = @modifier
+            @modifier = "public"
+            visit(node.arguments)
+            @modifier = modifier
+          end
+        end
+      end
+
+      sig { override.params(node: SyntaxTree::VCall).void }
+      def visit_vcall(node)
+        vcall_value = node.value.value
+
+        if ["private", "protected", "public"].include?(vcall_value)
```

Comment:
Let's put this array in a constant to avoid allocating it for every test we find.